atlasdb-dbkvs/src/test/java/com/palantir/atlasdb/keyvalue/dbkvs/impl/postgres/PostgresDdlTableTest.java

```diff
@@ -49,57 +48,67 @@
     private PostgresDdlTable postgresDdlTable;
     private static final ConnectionSupplier connectionSupplier = mock(ConnectionSupplier.class);
     private static final TableReference TEST_TABLE = TableReference.createFromFullyQualifiedName("ns.test");
-    private static final int NOW_MILLIS = 100;
+
+    private static final long NOW_MILLIS = 1000;
+    private static final long COMPACT_INTERVAL_MILLIS = 100;
+    private static final long SMALL_POSITIVE_FACTOR = 10;
 
     @Before
     public void setUp() {
         postgresDdlTable = new PostgresDdlTable(TEST_TABLE,
                 connectionSupplier,
-                ImmutablePostgresDdlConfig.builder().compactInterval(HumanReadableDuration.valueOf("10 ms")).build());
+                ImmutablePostgresDdlConfig.builder()
+                        .compactInterval(HumanReadableDuration.milliseconds(COMPACT_INTERVAL_MILLIS))
+                        .build());
     }
 
     @Test
-    public void shouldCompactIfVacuumWasNeverPerformed() throws Exception {
-        SqlConnection sqlConnection = setUpSqlConnection(null, null, null, null);
-        assertTrue(postgresDdlTable.checkIfTableHasNotBeenCompactedForCompactInterval());
+    public void shouldCompactIfVacuumWasNeverPerformedAndTheDbTimeIsLessThanCompactInterval() throws Exception {
+        SqlConnection sqlConnection = setUpSqlConnection(null, COMPACT_INTERVAL_MILLIS / SMALL_POSITIVE_FACTOR);
 
-        assertThatVacuumWasPerformed(sqlConnection);
+        assertThatVacuumWasPerformed(sqlConnection, true);
```

Comment:
(nit) It's strange to have `assertThatVacuumWasPerformed` that takes a boolean and `assertThatVacuumWasNotPerformed` that does not. Either rename the former and use it in both cases, or overload it so that you don't need to pass in a boolean.